CDS/js/youtube/youtube-auth.js

```diff
@@ -0,0 +1,97 @@
+var googleapis = require('googleapis');
+var OAuth2 = googleapis.auth.OAuth2;
+var fs = require('fs');
+var Promise = require('promise');
+var pem = require('pem');
+var Q = require('q');
+
+const SECRET_KEY_FILE_PATH = './privatekey.pem';
```

Comment:
The private key gets stored in this file, and the file then gets passed on to google.